nucypher/cli/commands/workers.py

```diff
@@ -0,0 +1,122 @@
+"""
+ This file is part of nucypher.
+
+ nucypher is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Affero General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ nucypher is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+import click
+
+try:
+    from nucypher.utilities.clouddeploy import CloudDeployers
+except ImportError:
+    CloudDeployers = None
+from nucypher.cli.utils import setup_emitter
+from nucypher.config.characters import StakeHolderConfiguration
+from nucypher.cli.commands.stake import group_staker_options, option_config_file, group_general_config
+
+
+def filter_staker_addresses(stakers, stakes):
+
+    staker_addresses = set()
+    for staker in stakers:
+        for stake in staker.stakes:
+            if stakes:
+                if not stake.staker_address in stakes:
+                    continue
```

Comment:
The third level of nesting here is unnecessary since stakers are 1:1 with workers.